stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -277,81 +277,83 @@ function rmul!(A::AbstractMatrix, transB::Transpose{<:Any,<:Diagonal})
 end
 
 # Elements of `out` may not be defined (e.g., for `BigFloat`). To make
-# `mul!(out, A, B)` work for such cases, `_scaledout` short-circuits
+# `mul!(out, A, B)` work for such cases, `out .*ₛ beta` short-circuits
```

Comment:
This doesn't render for me. Which character are you using here?

Reply:
It's `*_s` (for short-circuiting):

![image](https://user-images.githubusercontent.com/29282/63149207-4db36180-bfb8-11e9-9bfc-939886823c4b.png)

Maybe `*ˢ` (superscript `s`) or `*′` (prime)?

Or we can pick a non-standard character (maybe `⋇`?) from:

`* / ÷ % & ⋅ ∘ × |\\| ∩ ∧ ⊗ ⊘ ⊙ ⊚ ⊛ ⊠ ⊡ ⊓ ∗ ∙ ∤ ⅋ ≀ ⊼ ⋄ ⋆ ⋇ ⋉ ⋊ ⋋ ⋌ ⋏ ⋒ ⟑ ⦸ ⦼ ⦾ ⦿ ⧶ ⧷ ⨇ ⨰ ⨱ ⨲ ⨳ ⨴ ⨵ ⨶ ⨷ ⨸ ⨻ ⨼ ⨽ ⩀ ⩃ ⩄ ⩋ ⩍ ⩎ ⩑ ⩓ ⩕ ⩘ ⩚ ⩜ ⩞ ⩟ ⩠ ⫛ ⊍ ▷ ⨝ ⟕ ⟖ ⟗`

Reply:
We can also use normal function like `mulsc`.  The code would look like:

    mulsc.(A.diag .* in, alpha) .+ mulsc.(out, beta)

Reply:
According to wikipedia ⊙ is sometimes used for element-wise product.

Reply:
`⊙` is indeed the notation that my profs used for element-wise product.

Reply:
We need a non-standard binary operator and directly define the broadcasted behavior of it.  (The usecase here is to hijack the broadcasted behavior to implement short-circuiting at the level of vector-vector multiplication.)

So using `⊙` would mean to write `.⊙` which looks a bit strange as it sounds like doing "doubly" element-wise operation on nested objects.

Reply:
[TensorCore.jl](https://github.com/JuliaMath/TensorCore.jl)